core/deployment/src/main/java/io/quarkus/deployment/logging/LoggingResourceProcessor.java

```diff
@@ -79,11 +82,25 @@ void miscSetup(
     @BuildStep
     @Record(ExecutionTime.RUNTIME_INIT)
     void setupLoggingRuntimeInit(LoggingSetupRecorder recorder, LogConfig log, List<LogHandlerBuildItem> handlers,
-            List<LogConsoleFormatBuildItem> consoleFormatItems) {
-        final List<RuntimeValue<Optional<Handler>>> list = handlers.stream().map(LogHandlerBuildItem::getHandlerValue)
+            List<LogConsoleFormatBuildItem> consoleFormatItems, List<NamedLogHandlersBuildItem> namedLogHandlersBuildItems) {
+        final List<RuntimeValue<Optional<Handler>>> additionalHandlers = handlers.stream()
+                .map(LogHandlerBuildItem::getHandlerValue)
+                .collect(Collectors.toList());
+        final List<RuntimeValue<Optional<Formatter>>> possibleFormatters = consoleFormatItems.stream()
+                .map(LogConsoleFormatBuildItem::getFormatterValue).collect(Collectors.toList());
+        final List<RuntimeValue<Optional<Map<String, List<Handler>>>>> namedHandlers = namedLogHandlersBuildItems.stream()
```

Comment:
This is still wrong, it has to be a `Map<String, List<RuntimeValue<Optional<Handler>>>>` I think.  I'll review more after I get back from running an errand.

Reply:
Rather it has to be a `Map<String, RuntimeValue<Optional<Handler>>>`.  If there is more than one handler with a given name, we need to fail the build.  Then the map is passed to the recorder, so the `RuntimeValue` is not dereferenced until the proper time.